File: test/smartest_runner_test_ffi.mjs
// SPDX-FileCopyrightText: 2026 gleam_mutants contributors
// SPDX-License-Identifier: MIT OR Apache-2.0

export function block(milliseconds) {
  const wait = Math.max(0, Number(milliseconds));
  Atomics.wait(new Int32Array(new SharedArrayBuffer(4)), 0, 0, wait);
}

File: test/smartest_runner_test_ffi.erl
%% SPDX-FileCopyrightText: 2026 gleam_mutants contributors
%% SPDX-License-Identifier: MIT OR Apache-2.0

-module(smartest_runner_test_ffi).
-export([block/1]).

block(Milliseconds) ->
    timer:sleep(Milliseconds),
    nil.
